src/Weeder/Main.hs

```diff
@@ -104,9 +164,27 @@ parseCLIArguments = do
     pure CLIArguments{..}
 
 
+data Weed = Weed
+  { weedPath :: FilePath
+  , weedLoc :: RealSrcLoc
+  , weedDeclaration :: Declaration
+  , weedPrettyPrintedType :: Maybe String
+  }
+
+
+formatWeed :: Weed -> String
+formatWeed Weed{..} =
+  weedPath <> ":" <> show ( srcLocLine weedLoc ) <> ": "
+    <> case weedPrettyPrintedType of
+      Nothing -> occNameString ( declOccName weedDeclaration )
+      Just t -> "(Instance) :: " <> t
```

Comment:
I think this belongs in `Weeder`, rather than `Weeder.Main`. `Weeder.Main` is really about just being a driver for `Weeder`, and its interaction with the outside world (so I think exit codes are fine in `Weeder.Main`).

Reply:
I would say this is still directly related to interacting with the outside world since this only exists to define the appearance of the output, rather than relating to building the dependency graph.